src/main/java/uk/nhs/hee/tis/trainee/sync/service/HeeUserSyncService.java

```diff
@@ -60,13 +60,19 @@ public void syncRecord(Record heeUser) {
       throw new IllegalArgumentException(message);
     }
 
-    if (heeUser.getOperation().equals(DELETE)) {
-      repository.deleteById(heeUser.getTisId());
-    } else {
+    String name = heeUser.getData().get(HEE_USER_NAME);
+    Optional<HeeUser> heeUserOptional = findByName(name);
+    heeUserOptional.ifPresent(user -> {
+      repository.deleteById(user.getTisId());
+      log.info("Deleted HEE user {}.", user);
```

Comment:
Since the updated record may have other changes, we delete the existing one and then insert the new version.